flow/aggregates.py

```diff
@@ -8,10 +8,40 @@
 default aggregator produces individual jobs.
 """
 import itertools
+from abc import abstractmethod
 from collections.abc import Iterable, Mapping
 from hashlib import md5
 
 
+def _get_unique_function_id(func):
```

Comment:
Since this is a utility function which currently only serves the `aggregator` class, I think this can be a static method.

Reply:
Although this _could_ be a static method, we generally prefer to define helper methods over static methods in our code base. In this particular case, the code doesn't even have anything to do with the `aggregator` class, so calling it as `aggregator._get_unique_function_id(func)` would indicate an association between the class and function even though they're not related.

Reply:
Okay! But since this is not closely related to aggregation, should we put this private function inside the utils directory?

Reply:
Possibly. But keeping it in the same module seems best until we use it elsewhere.